compiler/AST/ForallStmt.cpp

```diff
@@ -389,16 +455,24 @@ BlockStmt* ForallStmt::build(Expr* indices, Expr* iterator,
     indices = new UnresolvedSymExpr("chpl__elidedIdx");
   checkIndices(indices);
 
-  if (!forall_intents)
-    forall_intents = new ForallIntents();
-
-  ForallStmt* fs = new ForallStmt(zippered, body, forall_intents);
+  ForallStmt* fs = new ForallStmt(zippered, body);
+
+  if (forall_intents) {
+    // transfer from 'forall_intents' into 'fs'
+    // todo: how to make the parser produce AList directly?
+    AList& fis = fs->forallIntents();
+    for (int i = 0; i < forall_intents->numVars(); i++)
+      fis.insertAtTail(new ForallIntent(forall_intents->fIntents[i],
+                                        forall_intents->fiVars[i],
+                                        forall_intents->riSpecs[i]));
+    delete forall_intents;
```

Comment:
Something (PR message, comments, TODOs, issues) needs to clearly say that we don't intend to have both ForallIntents and ForallIntent going forward. What will it take to get the parser to produce ForallIntent only? Could you retire ForallIntents in this PR? If you can, you should. I think it's confusing to have both, if we can avoid it.